src/structures/interfaces/TextBasedChannel.js

```diff
@@ -159,25 +159,41 @@ class TextBasedChannel {
   /**
    * Starts a typing indicator in the channel.
    * @param {number} [count] The number of times startTyping should be considered to have been called
+   * @returns {Promise}
    * @example
    * // Start typing in a channel
    * channel.startTyping();
    */
   startTyping(count) {
     if (typeof count !== 'undefined' && count < 1) throw new RangeError('TYPING_COUNT');
     if (!this.client.user._typing.has(this.id)) {
-      const endpoint = this.client.api.channels[this.id].typing;
-      this.client.user._typing.set(this.id, {
-        count: count || 1,
-        interval: this.client.setInterval(() => {
-          endpoint.post();
-        }, 9000),
+      const entry = {};
+      entry.promise = new Promise((resolve, reject) => {
```

Comment:
you can do
```js
const entry = {
  promise: ...
}
```

Reply:
That would cause the `entry` variable to be undefined in the promise' executor.

Reply:
sounds good